File: crates/graph-nexus-cli/tests/fixtures/receiver_types.cpp
class Apple {
public:
    void eat() {
        this->peel();
    }
    void peel() {}
};

class Banana : public Apple {
public:
    void eat() {
        Apple::eat();
    }
};

void useParam(Apple a) {
    a.eat();
}

void useLocal() {
    Apple a;
    a.eat();
    Apple* p = &a;
    p->eat();
}

void useNoAnnotation() {
    auto a = Apple();
    a.eat();
}

File: crates/graph-nexus-cli/tests/fixtures/receiver_types.swift
class Apple {
    func eat() {
        self.peel()
    }
    func peel() {}
}

class Banana: Fruit {
    override func eat() {
        super.eat()
    }
}

extension Apple {
    func slice() {
        self.peel()
    }
}

func useParam(a: Apple) {
    a.peel()
}

func useLocal() {
    let a: Apple = Apple()
    a.peel()
}

func useNoAnnotation() {
    let a = Apple()
    a.peel()
}
